### PR TITLE
Pin Rust Nightly to 2020-10-25

### LINKED ISSUE #137: Rust nightly 2020-10-25 broke build & is not supported
Looks like our CI started failing with the Rust nightly build from today. 
https://github.com/EmbarkStudios/rust-gpu/runs/1305181659

Maybe we should pin to a specific nightly version in `rust-toolchain` now when more people are working with the repo to prevent breakages as it is easier for more to be on different versions @khyperia ?

```
   Compiling rustc_codegen_spirv v0.1.0 (D:\a\rust-gpu\rust-gpu\rustc_codegen_spirv)
error[E0053]: method `add_counter_region` has an incompatible type for trait
   --> rustc_codegen_spirv\src\builder\mod.rs:218:5
    |
218 | /     fn add_counter_region(
219 | |         &mut self,
220 | |         _instance: Instance<'tcx>,
221 | |         _function_source_hash: u64,
...   |
225 | |         todo!()
226 | |     }
    | |_____^ expected `bool`, found `()`
    |
    = note: expected fn pointer `fn(&mut builder::Builder<'a, 'tcx>, rustc_middle::ty::Instance<'_>, _, CounterValueReference, CodeRegion) -> bool`
               found fn pointer `fn(&mut builder::Builder<'a, 'tcx>, rustc_middle::ty::Instance<'_>, _, CounterValueReference, CodeRegion)`
```

### PR DESCRIPTION
**Pull request by repi**
Fix #137

### CODE REVIEW

File: setup.bat [hunk 1]
```diff
@@ -1,3 +1,3 @@
 setlocal
 
-rustup toolchain install nightly --component rust-src rustc-dev llvm-tools-preview
+rustup toolchain install nightly-2020-10-24 --component rust-src rustc-dev llvm-tools-preview
```
Comment (bjorn3) [inline setup.bat:3]: Wouldn't `rustup component add rust-src rustc-dev llvm-tools-preview` automatically install the correct nightly?

Comment (repi) [reply setup.bat:3]: Probably, if it does read the `rust-toolchain` file, worth trying

**Comment by khyperia** [comment]
Oh, uh, didn't see this in my pile of inbox before I already pushed a fix to get it to compile on latest nightly.

**Comment by repi** [comment]
I can update this pinning to the date of the latest nightly

**Comment by khyperia** [comment]
sounds good! I don't want to r+ this because then it'd automerge, and fail because the CI run now is out of date, but, pseudo-r+ from me for pinning to latest nightly :stuck_out_tongue:

**Comment by repi** [comment]
Thx, will update this

**Comment by repi** [comment]
@khyperia rebased this to latest, updated to use 2020-10-25 instead and changed current setup scripts to just add the components in general to bjorn3's suggestion instead of to a specific nightly version. Makes it a bit easier.

So if you approve this one right now it should be merged automatically if tests pass

File: setup.bat [hunk 2]
```diff
@@ -1,3 +1,3 @@
 setlocal
 
-rustup toolchain install nightly-2020-10-24 --component rust-src rustc-dev llvm-tools-preview
+rustup component add rust-src rustc-dev llvm-tools-preview
```
Comment (bjorn3) [inline setup.bat:3]: nit: missing trailing newline

File: setup.sh [hunk 3]
```diff
@@ -1,2 +1,2 @@
 #!/usr/bin/env bash
-rustup toolchain install nightly --component rust-src rustc-dev llvm-tools-preview
+rustup component add rust-src rustc-dev llvm-tools-preview
```
Comment (bjorn3) [inline setup.sh:2]: nit: missing trailing newline

**Pull request merged**